components/provisioner/internal/api/mockresolver_test.go

```diff
@@ -0,0 +1,481 @@
+package api
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/kyma-incubator/compass/components/provisioner/pkg/gqlschema"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestResolver_ProvisionRuntime(t *testing.T) {
+	repository := make(map[string]RuntimeOperation)
+
+	resolver := NewMockResolver(repository)
+	runtimeID := "1234"
+
+	t.Run("Should return OperationID when runtime provisioning starts", func(t *testing.T) {
+		//given
+		ctx := context.Background()
+		input := &gqlschema.ProvisionRuntimeInput{}
+
+		//when
+		id, e := resolver.ProvisionRuntime(ctx, runtimeID, input)
+
+		//then
+		require.NoError(t, e)
+		require.NotEmpty(t, id)
+
+		//cleanup
+		resolver.repository = make(map[string]RuntimeOperation)
+	})
+
+	t.Run("Should return error when another operation is in progress", func(t *testing.T) {
+		//given
+		ctx := context.Background()
+		input := &gqlschema.ProvisionRuntimeInput{}
+
+		operation := RuntimeOperation{
```

Comment:
Why we couldn't just call resolver.ReconnectRuntimeAgent (or any other mutation). It would make the code shorter.